File: frontend/src/interfaces/RoomType.ts
import type { RoomInterface } from "./Room";

export interface RoomType {
  ID: number;
  CreatedAt: string;
  UpdatedAt: string;
  DeletedAt: string | null;

  RoomTypeName: string;
  RentalPrice: number;

  // Relations
  Rooms?: RoomInterface[];
}

File: backend/entity/RoomType.go
package entity

import "gorm.io/gorm"

type RoomType struct {
    gorm.Model
    RoomTypeName string  `json:"RoomTypeName"`
    RentalPrice  float64 `json:"RentalPrice"`

    // Relations
    Rooms []Room `json:"Rooms" gorm:"foreignKey:RoomTypeID"`
}//kuy2
